src/backend/api/Fusion.Resources.Domain/Queries/Departments/GetDepartmentHandlerBase.cs

```diff
@@ -14,53 +11,54 @@ namespace Fusion.Resources.Domain
+            this.rolesClient = rolesClient;
             this.lineOrgResolver = lineOrgResolver;
             this.profileResolver = profileResolver;
         }
 
         internal async Task ExpandDelegatedResourceOwner(QueryDepartment department, CancellationToken cancellationToken)
         {
-            var delegatedResourceOwners = await db.DepartmentResponsibles
-                            .Where(r => r.DepartmentId == department.DepartmentId)
-                            .ToListAsync(cancellationToken);
-
+            var delegatedResourceOwners = await rolesClient.GetRolesAsync(q => q
+                   .WhereScopeValue(department.DepartmentId)
+                   .WhereScopeType("OrgUnit")
+                   .WhereRoleName(AccessRoles.ResourceOwner))
+               ;
             await ResolveDelegatedOwners(department, delegatedResourceOwners);
         }
 
         internal async Task ExpandDelegatedResourceOwner(List<QueryDepartment> departments, CancellationToken cancellationToken)
         {
-            var departmentIds = departments.Select(x => x.DepartmentId).ToArray();
+            var allDelegatedResourceOwners = (await rolesClient.GetRolesAsync(q => q
+             .WhereScopeType("OrgUnit")
+             .WhereRoleName(AccessRoles.ResourceOwner))).ToList();
 
-            var query = await db.DepartmentResponsibles
-                            .Where(r => departmentIds.Contains(r.DepartmentId))
-                            .ToListAsync(cancellationToken);
-
-            var delegatedMap = query.ToLookup(x => x.DepartmentId);
+            if (!allDelegatedResourceOwners.Any())
+                return;
 
             foreach (var department in departments)
             {
-                if (delegatedMap.Contains(department.DepartmentId))
-                    await ResolveDelegatedOwners(department, delegatedMap[department.DepartmentId]);
+                var departmentResourceOwners = allDelegatedResourceOwners.Where(x => x.Scope!.Values.Contains(department.DepartmentId));
+                await ResolveDelegatedOwners(department, departmentResourceOwners);
             }
         }
 
-        private async Task ResolveDelegatedOwners(QueryDepartment department, IEnumerable<DbDepartmentResponsible> delegatedResourceOwners)
+        private async Task ResolveDelegatedOwners(QueryDepartment department, IEnumerable<FusionRoleAssignment> delegatedResourceOwners)
         {
             if (delegatedResourceOwners is null) return;
-         
+
             var resolvedProfiles = await profileResolver
-                .ResolvePersonsAsync(delegatedResourceOwners.Select(p => new PersonIdentifier(p.ResponsibleAzureObjectId)));
+                .ResolvePersonsAsync(delegatedResourceOwners.Select(p => new PersonIdentifier(p.Person!.AzureUniqueId)));
 
             department.DelegatedResourceOwners = resolvedProfiles
                 .Where(res => res.Success)
                 .Select(res => res.Profile!)
```

Comment:
I guess resolved profile may me null :-o
Try to run a search in CI using TDI as search arg.

Reply:
I tried Localy and it throws an error before it comes into here, in the LineOrgExtension.cs file. 
![image](https://user-images.githubusercontent.com/3900367/194003587-59369be3-3bcc-4ce8-871e-d1726c77d4ba.png)

i do not know the adress to the ci  enivronment. 
I have the prod link: https://fap-resources-api-fprd.azurewebsites.net 
but do not figure out the ci link.   as replacing fprd with ci does not work.

Reply:
Test environments have the following pattern:
https://resources-api.**<env>**.fusion-dev.net/

Reply:
Looks like same error in CI to 

![image](https://user-images.githubusercontent.com/3900367/194022519-30a43459-3ece-4313-bda0-7a320cf62c17.png)

Reply:
This is the error you should resolve.

Reply:
Should have been fixed now.  
returns a []  empty list, and ut appeasr to works as desired, no errors are thrown based on testing